src/Renderer.tsx

```diff
@@ -38,24 +40,38 @@ type Fallback = 'ignore' | 'warning' | 'passthru' | ComponentType<{ node: Node }
 
 interface Props<N extends Node | Node[]> {
     children?: ReactNode;
+    coverageEntries?: Record<number, CoverageEntry>;
     defaultComponents?: boolean;
     defaultFallback?: Fallback;
     nodes: N;
+    renderDate?: (date: string) => ReactNode;
     transformations?: Transformation[];
 }
 
+interface RenderProps<T extends Node> {
+    children?: ReactNode;
+    node: T;
+}
+
 export function Renderer<N extends Node | Node[]>({
     children,
+    coverageEntries = {},
     defaultComponents = true,
     defaultFallback = 'warning',
     nodes,
+    renderDate = defaultRenderDate,
     transformations = Object.values(Transformations),
 }: Props<N>) {
     const transformedNodes = applyTransformations<Node>(
         Array.isArray(nodes) ? nodes : [nodes],
         transformations,
     );
 
+    function renderCoverageNode(props: RenderProps<CoverageNode>) {
+        const coverage: CoverageEntry | undefined = coverageEntries[props.node.coverage.id];
+        return <Elements.Coverage coverage={coverage} node={props.node} renderDate={renderDate} />;
+    }
```

Comment:
In order to reduce the amount of boilerplate code required in each theme, I've implemented the coverage node "enrichment" inside the renderer itself.

I know it's not totally ideal as the renderer doesn't need to know about this stuff, but at the same time considering how many themes we have, it will lead to less bugs and easier time implementing.

I also know we do this outside of the renderer for story bookmarks, but I think we could also move it here eventually, unless you think it's not a good idea.

With this approach, we can provide the referenced stories and coverage to the renderer directly and would not need to use context like [we do in Bea](https://github.com/prezly/theme-nextjs-bea/blob/7b2fa2bce294de20b3fc6cdc17109c6a11ab222b/components/ContentRenderer/ContentRenderer.tsx#L91-L95) for example.

What do you think?